File: IT/2/Input-Output/Esercizi/Java/Esercizio12.java
/* file Esercizio12.java */

import java.io.Console;
import java.io.File;
import java.io.FileWriter;
import java.io.BufferedWriter;
import java.io.IOException;

public class Esercizio12 {
	public static void main(String [] args) {
		System.out.println("Scrivi una serie di parole. Invia la stringa vuota per uscire.");
		Console console = System.console();

		try {
			File doc = new File("file.txt");
			FileWriter writer = new FileWriter(doc);
			BufferedWriter buffer = new BufferedWriter(writer);

			while (true) {
				String s = console.readLine("Scrivi una parola: ");

				if (s.isEmpty()) {
					break;
				}

				buffer.write(s);
				buffer.write(System.lineSeparator());
			}

			buffer.close();

			System.out.println("File \"" + doc.getName() + "\" salvato su disco.");
		}
		catch (IOException e) {
			System.out.println(e);
		}
	}
}

File: IT/2/Input-Output/Esercizi/Python/esercizio12.py
# file esercizio12.py

from __future__ import print_function
if hasattr(__builtins__, 'raw_input'): input = raw_input
from os import linesep

print("Scrivi una serie di parole. Invia la stringa vuota per uscire.")

doc = open("file.txt", "w")

while (True):
	s = input("Scrivi una parola: ")

	if not s:
		break

	doc.write(s + linesep)

doc.close()

print("File \"{}\" salvato su disco.".format(doc.name))

